### DISCUSSION TITLE
Ariston Genus One 24 + Cube S

### DISCUSSION THREAD

**Discussion by VoyteckPL** [john30/ebusd#1214] [General]
I use HA with ebusd addon on RPI4.
I loaded ariston.csv by komw. Maybe half of parameters is available. How can I view (sniffing?) all transmitted parameters and compare it with real values on the heater?
I can't see the slope, heating offset etc.
Please help :)

**Comment by JonesGW-MGD** [comment]
Briefly, from the terminal line of the add-on instance (various ways to get there, I use) - 

```
docker exec -it `docker ps | grep ebusd | awk '{print $1}'` /bin/bash
```

You can try
```
ebusctl grab
```
then reviewing the results (after waiting some time for the messages to be present on the bus - alternatively you can 'make' messages appear by using your physical control devices), using
```
ebusctl grab result 
```
using 

```
ebusctl grab result decode
```
iterates through the various known base data types and each valid position of the master and slave data parts

You should find a reasonable amount of threads on this forum and the ebusd configuration files GitHub page to using 'grab'.

https://github.com/john30/ebusd/wiki/HowTos

**Comment by VoyteckPL** [comment]
Thanks it is working!
The problem is how to get it to .csv somehow. I mean all known 4 digit codes so I can see it.

**Reply by JonesGW-MGD** [reply]
I don't have an Ariston so can't really give you any clues - But if they are 'known', they are probably going to be in your version of the csv from komw.

You can try to decode the unknown ones yourself - by changing controls on the boiler or your remote controller if you have one and reviewing the messages, but that may prove quite difficult. (See that how-to link I posted). You'll probably have to invest quite a lot of time trying to decode the messages, and then understand in some depth the format of the message lines in the csv to add or amend - probably starting here: https://github.com/john30/ebusd/wiki/4.1.-Message-definition

You can maybe first start by searching for 'Ariston' on this forum and the config files forum - https://github.com/john30/ebusd-configuration/issues

Maybe other people have found some things that haven't made it to the komw csv file?  

One thing I found, although my boiler is Vaillant so maybe a wider community, is that even though ebusd loaded a default file, I tried the values from other files in trial and error mode for some unknown messages - that worked actually in half a dozen situations in fact.  You could try the same approach - my advice here would be to be clear about which missing values would actually be useful to you rather than just trying to fill all the gaps!

Also, final piece of advice, if you have some remote control device - such as the Cube - then often you can control the boiler through the controller by finding csv files that work for the controller, rather than focussing on the boiler.

Good luck!

**Reply by JonesGW-MGD** [reply]
Do you use this file for the Cube?  Maybe that will give you some more data?  Unless it doesn't use ebus to communicate with the boiler

https://github.com/john30/ebusd-configuration/issues/27#issuecomment-583498444

**Reply by VoyteckPL** [reply]
Thanks for support! I use this one:
https://github.com/komw/ariston-bus-bridgenet-ebusd/blob/master/configuration/ariston.csv

I'm not sure how to get for example heat slope and heat offset. The values are in .csv but I don't see them in HA.

**Reply by JonesGW-MGD** [reply]
what happens if you do
```
ebusctl read heat_slope 
```
If it doesn't return a value then potentially that .csv entry doesn't work for your Genus? 

if that returns a value, can you see a heat_curve topic in MQTT (I tend to use MQTT Explorer as a client to independently of HA monitor MQTT - it takes about 5mins to set up - I use it on my normal workstation, not the pi, pointing to the MQTT instance in the setup and is pretty useful)

If you can't see a topic your default mqtt config may be filtering it out.  You can try starting the ebusd add-on with 

```
--mqttvar="filter-name="
```
in the custom command line options section of the ebusd add-on configuration tab.

If you can see the topic fine and the sensor is still missing in HA, I guess you could try manually creating a sensor to that topic, not sure why the auto-discovery wouldn't have worked though

**Comment by VoyteckPL** [comment]
I use MQTT Explorer :) I added --mqttvar="filter-name=" but nothing changed.

Here is what I see:
![image](https://github.com/john30/ebusd/assets/10666159/7448a016-7743-4fbd-b324-4406b2c5b0da)

Hold on! I can see one extra value now!
Wow!
pump_max_pwm = { "0": {"name": "", "value": 99}}
pump_min_pwm = { "0": {"name": "", "value": 55}}

**Reply by JonesGW-MGD** [reply]
Try forcing a read of heat_slope with the ebusctl command as above
